iib/workers/tasks/build.py

```diff
@@ -496,6 +497,65 @@ def _opm_index_add(
 
     with set_registry_token(overwrite_from_index_token, from_index):
         run_cmd(cmd, {'cwd': base_dir}, exc_msg='Failed to add the bundles to the index image')
+    
+    # add properties to bundles in the index
+    if add_properties:
+        maxOpenShiftVersionProperty = {
+            "type": "olm.maxOpenShiftVersion",
+            "value": "4.8",
+            "operatorbundle_version": "",
+            "operatorbundle-path": "",
+        }
+        for bundle in bundle_str.split(','):
+            if _sdk_is_valid(bundle):
+                _add_property_to_index(base_dir, bundle, maxOpenShiftVersionProperty)
+
+
+@retry(attempts=2, wait_on=IIBError, logger=log)
+def _add_property_to_index(base_dir, bundle, property):
+    import sqlite3
+    """
+    Add a property to the index
+
+    :param str base_dir: the base directory where the database and index.Dockerfile are created
+    :param str bundle: the bundle path
+    :param dict property: a dict representing a property to be added to the index.db for a given bundle
+    :raises IIBError: if the sql insertion fails
+    """
+
+    insert = "INSERT INTO properties (type, value, operatorbundle_version, operatorbundle_path) VALUES (?, ?, ?, ?);"
+    dbFile = base_dir + "/database/index.db"
+    con = sqlite3.connect(dbFile)
+    # Get version for bundle
+    cur = con.cursor()
+    cur.execute("SELECT version FROM operatorbundle WHERE bundlepath=?", (bundle,))
+    rows = cur.fetchall()
+    if len(rows) != 1:
+        raise IIBError("Error getting version for bundle")
+    property["operatorbundle_version"] = rows[0][0]
+    property["operatorbundle_path"] = bundle
+    cur.close()
+    # Insert property
+    con.execute(insert, (property["type"], property["value"], property["operatorbundle_version"], property["operaotorbundle_path"]))
+    con.close()
+
+
+def _sdk_is_valid(bundle):
+    cmd = [
+        'operator-sdk',
+        'bundle',
+        'validate',
+        bundle,
+        '--select-optional',
+        'name=community',
+        '--output=json-alpha1'
+    ]
+    try:
+        run_cmd(cmd, {}, exc_msg='operator-sdk command failed')
+        return True
+    except IIBError:
+        # TODO: check that the error is related to v1beta1 api removal
```

Comment:
we can process the output which should look something like this:
```
ERRO[0001] Error: Value : (memcached-operator.v0.0.1) csv.Annotations not specified olm.maxOpenShiftVersion for an OCP version < 4.9. This annotation is required to prevent the user from upgrading their OCP cluster before they have installed a version of their operator which is compatible with 4.9. This bundle is using APIs which were deprecated and removed in v1.22. More info: https://kubernetes.io/docs/reference/using-api/deprecation-guide/#v1-22 which are no longer supported on 4.9. Migrate the API(s) for CRD: (["memcacheds.cache.example.com"]),ClusterRole: (["memcached-operator-metrics-reader"]), or use the annotation 
```

Reply:
will tackle that tomorrow

Reply:
Yep. I think we can check if the errors[] has an string ` not specified olm.maxOpenShiftVersion` to return true here.